apps/system/js/about_service_workers_proxy.js

```diff
@@ -0,0 +1,121 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global IACHandler */
+
+/* exports AboutServiceWorkersProxy */
+
+/**
+ * On B2G we cannot directly show the content of about:* pages with access to
+ * privileged code. We already have an about:serviceworkers chrome page showing
+ * the list of installed service workers on Desktop. Unfortunately we cannot
+ * access it from Gaia. To access this information, we need to query the
+ * platform from the System app. And to show it we chose to add a new panel
+ * inside the existing "Developer" panel in the Settings app. The communication
+ * with the Settings app is done via Inter App Communication API.
+ *
+ * This module acts as a proxy between the Settings app and the platform.
+ * The high level picture is the following:
+ *
+ *      Settings App
+ *          |
+ *          |- IAC
+ *          |
+ *      System App
+ *          |
+ *          |- Chrome/Content events
+ *          |
+ *        Gecko (ServiceWorkersManager accessed from a B2G dedicated component)
+ */
+
+'use strict';
+
+const ASW_CHROME_EVENT = 'mozAboutServiceWorkersChromeEvent';
+const ASW_CONTENT_EVENT = 'mozAboutServiceWorkersContentEvent';
+
```

Comment:
let's put this in IIEF
(function(exports) {
  exports.AboutServiceWorkersProxy = AboutServiceWorkersProxy;
}(window));